### PR TITLE
Analyze Middleware directory

### PR DESCRIPTION
**Pull request by greg0ire**
Spotted while working on https://github.com/doctrine/DoctrineBundle/pull/1767

### CODE REVIEW

File: psalm.xml.dist [hunk 1]
```diff
@@ -37,6 +38,18 @@
...
+                <!-- Random package should not call this class from the Doctrine bridge, but we are in the Doctrine bundle here -->
+                <referencedClass name="Symfony\Bridge\Doctrine\Middleware\Debug\Driver"/>
+            </errorLevel>
+        </InternalClass>
+        <InternalMethod>
+            <errorLevel type="suppress">
+                <!-- Random package should not call this method from the Doctrine bridge, but we are in the Doctrine bundle here -->
+                <referencedMethod name="Symfony\Bridge\Doctrine\Middleware\Debug\Driver::__construct"/>
+            </errorLevel>
+        </InternalMethod>
```
Comment (ostrolucky) [inline psalm.xml.dist]: do these not work as `@psalm-suppress`?

Comment (SenseException) [reply psalm.xml.dist]: We do have a lot `psalm-suppres` in the code and we do have a lot `suppress` in the psalm config. Is there a pattern in DoctrineBundle when to use one or the other?

Comment (ostrolucky) [reply psalm.xml.dist]: Pattern is that we put stuff into config that does not work with inline suppress. There are plenty of such cases when psalm-suppress doesn't have an effect but putting it to config does.

Comment (greg0ire) [reply psalm.xml.dist]: > do these not work as `@psalm-suppress`?

I'll try, it's just that I forgot that we were doing things like this in that repository.

Comment (greg0ire) [reply psalm.xml.dist]: It works.

**Review by ostrolucky — APPROVED** [review]

**Pull request merged**